backends/kv/writer.go

```diff
@@ -61,14 +63,46 @@ func (kv *Backend) Write(request *frames.WriteRequest) (frames.FrameAppender, er
 		return nil, err
 	}
 
+	var schema v3ioutils.V3ioSchema
+	schema, schemaError := v3ioutils.GetSchema(tablePath, container)
+
+	// Ignore 404 error, since it makes sense there is no schema yet.
+	tableAlreadyExists := true
+	if err != nil {
+		if errorWithStatus, ok := err.(*v3ioerrors.ErrorWithStatusCode); !ok || errorWithStatus.StatusCode() != http.StatusNotFound {
+			return nil, err
+		} else {
+			tableAlreadyExists = false
+		}
+	}
+
+	switch request.SaveMode {
+	case frames.Overwrite:
+		// If this is the first time we writing to the table, there is nothing to delete.
+		if tableAlreadyExists {
+			err = v3ioutils.DeleteTable(kv.logger, container, tablePath, "", kv.numWorkers, true)
+			if err != nil {
+				return nil, fmt.Errorf("error occured while deleting the table, err: %v", err)
```

Comment:
Better say which table in the error message (as you did in case of ErrorIfExists).